code/modules/food_and_drinks/machinery/smartfridge.dm

```diff
@@ -34,39 +44,153 @@
 			for(var/i in 1 to amount)
 				load(new typekey(src))
 
+/obj/machinery/smartfridge/Move(atom/newloc, direct, glide_size_override, update_dir)
+	var/turf/old_loc = loc
+	. = ..()
+	move_update_air(old_loc)
+
+/obj/machinery/smartfridge/can_be_unfasten_wrench(mob/user, silent)
+	if(welded_down)
+		to_chat(user, span_warning("[src] is welded to the floor!"))
+		return FAILED_UNFASTEN
+	return ..()
+
+/obj/machinery/smartfridge/set_anchored(anchorvalue)
+	. = ..()
+	if(!anchored && welded_down) //make sure they're keep in sync in case it was forcibly unanchored by badmins or by a megafauna.
+		welded_down = FALSE
+	can_atmos_pass = anchorvalue ? ATMOS_PASS_NO : ATMOS_PASS_YES
+	air_update_turf(TRUE, anchorvalue)
+
+/obj/machinery/smartfridge/welder_act(mob/living/user, obj/item/tool)
+	..()
+	if(istype(src, /obj/machinery/smartfridge/drying_rack))
+		return FALSE
+	if(welded_down)
+		if(!tool.tool_start_check(user, amount=1))
+			return TRUE
+		user.visible_message(span_notice("[user.name] starts to cut the [name] free from the floor."), \
+			span_notice("You start to cut [src] free from the floor..."), \
+			span_hear("You hear welding."))
+		if(!tool.use_tool(src, user, delay=100, amount=1, volume=100))
+			return FALSE
+		welded_down = FALSE
+		to_chat(user, span_notice("You cut [src] free from the floor."))
+		return TRUE
+	if(!anchored)
+		to_chat(user, span_warning("[src] needs to be wrenched to the floor!"))
+		return TRUE
+	if(!tool.tool_start_check(user, amount=1))
+		return TRUE
+	user.visible_message(span_notice("[user.name] starts to weld the [name] to the floor."), \
+		span_notice("You start to weld [src] to the floor..."), \
+		span_hear("You hear welding."))
```

Comment:
```suggestion
	user.visible_message(
		span_notice("[user.name] starts to weld the [name] to the floor."),
		span_notice("You start to weld [src] to the floor..."),
		span_hear("You hear welding."),
	)
```

native procs (as in, those without macro wrappers like `AddComponent()`) don't need backslashes in current DM and can (and should) be multi-lined like any list.